heidrun/smithsonian.rb

```diff
@@ -0,0 +1,446 @@
+    #   <place label=""Origin"">[value] *Duplicate values should be ignored."
+    spatial :class => DPLA::MAP::Place,
+            :each => record.if
+                           .field('indexedStructured', 'geoLocation')
+                           .fields('L2', 'L3', 'L4', 'L5', 'points', 'Other')
+                           .else { |r| r.field('freetext', 'place') },
+            :as => :place do
+      providedLabel place
+      lat place.field('point', 'latitude')
+      long place.field('point', 'longitude')
+    end
+
+    # dcterms:publisher
+    #   <freetext category="publisher" label="publisher">
+    publisher :class => DPLA::MAP::Agent,
+              :each => record.field('freetext', 'publisher')
+                             .match_attribute(:label, 'Publisher'),
+              :as => :publisher do
+      providedLabel publisher
+    end
+    # assuming a typo in the spec, seeing - label="Publisher"
+
+    # dc:rights
+    #   <media ... rights="[value]">
+    #     OTHERWISE <freetext category="creditLine" label="Credit line">;
+    #   <freetext category="objectRights" label="Rights">
+    rights record.map(&rights_map).flatten
+
+    # dcterms:subject
+    #   <freetext category="topic" label="Topic">;
+    #   <freetext category="culture" label="Nationality">;
+    #   <topic>;<name>;<culture>;<tax_kingdom>; <tax_phylum>; <tax_division>;
+    #   <tax_class>; <tax_order>; <tax_family>;  <tax_sub-family>;
+    #   <scientific_name>; <common_name>;<strat_group>; <strat_formation>;
+    #   <strat_member>
+    #
+    #   n at least one record
+    #   http://content9.qa.dp.la/qa/compare?id=825ca339b107da76b17a1ba49f3e92fe
+    #   there are @label values of "subject" and "event" which seem like they
+    #   should also be mapped to Subject.
+    subject :class => DPLA::MAP::Concept,
+            :each => record.map(&subject_map).flatten,
+            :as => :subject do
+      providedLabel subject
+    end
+
+    # dcterms:temporal
+    #   <date>; <geo_age-era>; <geo_age-system>; <geo_age-series>;
+    #   <geo_age-stage>
+    temporal :class => DPLA::MAP::TimeSpan,
+             :each => record.field('indexedStructured')
+                            .fields('date', 'geo_age-era', 'geo_age-system',
+                                    'geo_age-series', 'geo_age-stage'),
+             :as => :time do
+      providedLabel time
+    end
+
+    # dcterms:title
+    #   <title label="Title"> ;
+    #   <title label="Object Name">;
+    #   <title label="Title (Spanish)">
+    title record.field('descriptiveNonRepeating', 'title')
+                .match_attribute(:label) { |label|
+                  ['Title', 'Object Name', 'Title (Spanish)'].include?(label)
+                }
+
+    # dcterms:type
+    #   <online media type>. If it does not match a DCMI type, map it to image
+    dctype record.field('indexedStructured', 'online_media_type')
+    # TODO: should defaulting to 'Image' be handled as an enrichment? - JB
```

Comment:
Yes, I think it's best to do this in enrichment. We have enrichments that can handle this in place already (`MoveNonDCMIType` and `DCMITypeEnforcer`).